reconcile/utils/oc.py

```diff
@@ -668,6 +682,252 @@ def _run_json(self, cmd, allow_not_found=False):
         return out_json
 
 
+class OCNative(OCDeprecated):
+    def __init__(self, server, token, jh=None, settings=None,
+                 init_projects=False, init_api_resources=False,
+                 local=False):
+        super().__init__(server, token=token, jh=jh, settings=settings,
+                         init_projects=False, init_api_resources=False,
+                         local=local)
+        if server is not None and len(server) > 0:
+            self.client = self._get_client(server, token)
+            self.api_kind_version = self.get_api_resources()
+        else:
+            init_api_resources = False
+            init_projects = False
+
+        self.object_clients = {}
+        self.init_projects = init_projects
+        if self.init_projects:
+            self.projects = \
+                [p['metadata']['name']
+                 for p
+                 in self.get_all('Project.project.openshift.io')['items']]
```

Comment:
do you know if this works in OpenShift v3?

Reply:
I'm not sure.  I've run this against all clusters in app-interface (ie not specified specific cluster) and it hasn't thrown any errors, but I haven't made changes to anything that would apply to a v3 cluster to test it reads everything correctly.  I should do that.

Reply:
tested and it works